File: NoxPrime.cpp
/*
 * NoxPrime.cpp
 *
 *  Created on: 8.8.2025
 *      Author: Perceus Adalian
 */

#include <iostream>
#include <cstdlib>
#include <ctime>
#include <string>

int main()
{
	bool exitStatus = true;

	while (exitStatus)
	{
		int renderChars;

		std::cout << "Hello. Thank you for choosing to generate your password with NoxPrime." << std::endl;
		std::cout << "Please choose a length for your password: ";
		std::cin >> renderChars;

		if (std::cin.fail())
		{
			std::cin.clear();
			std::cin.ignore(10000);
			std::cout << "Invalid input; was expecting an integer." << std::endl;
			continue;
		}
		else if (renderChars <= 0 || renderChars > 20)
		{
			std::cout << "Invalid input; was expecting 0<=value<=20" << std::endl;
			continue;
		}

		srand(static_cast<unsigned int>(time(0)));
		std::string validCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()";
		std::string passwordString = "";
		for (int i = 0; i < renderChars; i++)
		{
			int randomIndex = rand() % validCharacters.length();
			passwordString += validCharacters[randomIndex];
		}

		std::cout << "Password Generated with " << renderChars << " characters: " << passwordString << std::endl;
		exitStatus = false;
	}
	return 0;
}

File: NoxPrime.py
import random
import string
import time

def animatePrint(string, timeSeconds):
    if not string:
        return
    delay = timeSeconds / len(string)
    for ch in string:
        print(ch, end='', flush=True)
        time.sleep(delay)
    print()

def main():
    animatePrint('ν/NOX: Password Generator', 1.25)
    
    while True:
        
        try:
            renderChars = int(input('Enter a value to generate a password: '))
        except ValueError:
            print('Invalid input; was expecting an integer.')
            continue
        if renderChars <= 0 or renderChars > 100:
            print('Invalid input; was expecting 0 < VALUE <= 100')
            continue
    
        validCharacters = string.ascii_letters + string.digits + f'!@#$%^&*()'
        passwordString = ''
    
        for i in range(renderChars):
            randomIndex = random.randint(0, len(validCharacters) - 1)
            index = randomIndex+0
            passwordString += validCharacters[index]

        animatePrint(f'Generating Password w/ {renderChars} Characters:',1.25)
        animatePrint(passwordString,2)
        
        continueGen = input('|$| Continue? [0/y]es | [1/n]o : ')
        if continueGen == 'y' or continueGen == 0:
            continue
        elif continueGen == 'n' or continueGen == 1:
            break
        break
    return 0                
           
main()